src/components/Items/Scorecard/Scorecard.vue

```diff
@@ -0,0 +1,252 @@
+import IconButton from "@/components/UI/Buttons/IconButton.vue";
+
+const confetti = require("canvas-confetti");
+const confettiCanvas = document.getElementById("confetticanvas");
+const confettiHandler = confetti.create(confettiCanvas, { resize: true });
+
+export default {
+  name: "Scorecard",
+  components: {
+    CircularProgress,
+    IconButton,
+  },
+  props: {
+    metrics: {
+      // details of all the metrics to show
+      // format of one metric --
+      // {
+      //   name: "METRIC_NAME",
+      //   icon: {
+      //     source: "SVG_NAME.svg",
+      //     color: "TAILWIND_CLASSES",
+      //   },
+      //   value: VALUE_TO_SHOW,
+      //   description: DESCRIPTION_TO_SHOW,
+      // },
+      default: () => [],
+      type: Array,
+    },
+    progressPercentage: {
+      // progress to show on the progress bar in %
+      default: 0,
+      type: Number,
+    },
+    greeting: {
+      // greeting of the scorecard
+      default: "",
+      type: String,
+    },
+    showScorecard: {
+      // indicator of when the scorecard has popped up
+      default: false,
+      type: Boolean,
+    },
+    plioTitle: {
+      // plio's title
+      default: "",
+      type: String,
+    },
+    isPortrait: {
+      default: false,
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      localProgressPercentage: 0, // local value of progress
+      showConfetti: false, // to show confetti or not
+      progressNumberIndicator: {
+        // what to show in between the progress bar
+        enabled: true,
+        additionalText: this.$t("player.scorecard.metric.description.progress"),
+      },
+      // variable to store the animation frame request
+      // used when confetti popping has to be stopped
+      animationFrameRequest: null,
+      innerWidth: window.innerWidth, // variable to hold the width of window
+      reRenderKey: 0, // a key to re-render a component
+    };
+  },
+  watch: {
+    showScorecard(value) {
+      if (value) {
+        // if scorecard pops up then wait some time to update the
+        // `localProgressPercentage`. This makes the progress bar animate
+        setTimeout(() => {
+          this.localProgressPercentage += this.progressPercentage;
+          this.showConfetti = true;
+        }, 500);
+
+        // Also, throw some confetti in there
+        this.throwConfetti();
+      } else {
+        // if scorecard is not visible anymore, reset things
+        this.localProgressPercentage = 0;
+        this.showConfetti = false;
+      }
+    },
+  },
+  created() {
+    // reactively handle screen resizing
+    window.addEventListener("resize", this.handleScreenSizeChange);
+  },
+  mounted() {},
+  unmounted() {
+    window.removeEventListener("resize", this.handleScreenSizeChange);
+  },
+  computed: {
+    circularProgressRadius() {
+      // reactively control the radius of the circular progress bar
+      // according to the screen width
```

Comment:
let's use this in general for function/computed variable docstrings - `/** */` - and this is added above the function - not inside it.